src/Tools/ExternalAccess/Xaml/External/XamlRequestHandlerBase.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.Xaml;
+
+internal abstract class XamlRequestHandlerBase<TRequest, TResponse> : ILspServiceDocumentRequestHandler<TRequest, TResponse>
+{
+    private readonly IXamlRequestHandler<TRequest, TResponse>? _xamlRequestHandler;
+
+    public XamlRequestHandlerBase(IXamlRequestHandler<TRequest, TResponse>? xamlRequestHandler)
+    {
+        _xamlRequestHandler = xamlRequestHandler;
+    }
+
+    public bool MutatesSolutionState => _xamlRequestHandler?.MutatesSolutionState ?? false;
```

Comment:
are there xaml requests that ever mutate the solution?

Reply:
No, but since they are required to implement it as part of `IMethodHandler` I figured I would respect it.